app/src/main/java/com/arctouch/publictransportation/activity/RouteListActivity.java

```diff
@@ -0,0 +1,193 @@
+import com.arctouch.publictransportation.volley.VolleySingleton;
+import com.arctouch.publictransportation.volley.endpoints.RoutesVolley;
+
+import java.util.List;
+
+public class RouteListActivity extends AppCompatActivity implements VolleyListener<RoutesReturn> {
+    private View mRecyclerView;
+    private ProgressBar mProgressBar;
+    private TextView mEmptyList;
+
+    private boolean mMultiplePane;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_route_list);
+
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        toolbar.setTitle(getTitle());
+
+        EditText mTextSearch = (EditText) findViewById(R.id.textSearch);
+        mTextSearch.setOnEditorActionListener(searchAction);
+
+        this.mProgressBar = (ProgressBar) findViewById(R.id.progressBar);
+        this.mEmptyList = (TextView) findViewById(R.id.emptyList);
+
+        this.mRecyclerView = findViewById(R.id.routeList);
+        assert mRecyclerView != null;
+
+        // Only show in large-screen layouts
+        if (findViewById(R.id.stopListContainer) != null) {
+            mMultiplePane = true;
+        }
+    }
+
+    private TextView.OnEditorActionListener searchAction = new TextView.OnEditorActionListener() {
+        @Override
+        public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
+            if (actionId == EditorInfo.IME_ACTION_SEARCH) {
+                InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
+                imm.hideSoftInputFromWindow(v.getWindowToken(), 0);
+                loadRoutes(v.getText().toString());
+                return true;
+            }
+            return false;
+        }
+    };
+
+    private void loadRoutes(String search) {
+        this.mProgressBar.setVisibility(View.VISIBLE);
+        this.mEmptyList.setVisibility(View.GONE);
+        RequestQueue requestQueue = VolleySingleton.getInstance(this).getRequestQueue();
+        RoutesVolley routesVolley = new RoutesVolley(this, new StopNameParam(search), this);
+        requestQueue.add(routesVolley.getmRequest());
+    }
+
+    @Override
+    public void onVolleySuccess(RoutesReturn routesReturn) {
+        this.mProgressBar.setVisibility(View.GONE);
+        this.mEmptyList.setVisibility(routesReturn.getRows().size() > 0 ? View.GONE : View.VISIBLE);
+        setupRecyclerView((RecyclerView) this.mRecyclerView, routesReturn);
+    }
+
+    @Override
+    public void onVolleyError(String error) {
+        this.mProgressBar.setVisibility(View.GONE);
+        this.mEmptyList.setVisibility(View.GONE);
+        AlertDialog dialog = createErrorDialog(error);
+        dialog.show();
+    }
+
+    private AlertDialog createErrorDialog(String message) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
```

Comment:
Since you're using a builder, there's no reason not to have a single statement:

```
return new AlertDialog.Builder(this)
                .setMessage(message)
                .setTitle(R.string.exception_retrieve_values_title)
                .setPositiveButton(R.string.ok, new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                        dialog.dismiss();
                    }
                })
                .create();